python/cudf/cudf/core/dataframe.py

```diff
@@ -1252,7 +1252,24 @@ def dtypes(self) -> pd.Series:
         string              object
         dtype: object
         """
-        return pd.Series(dict(self._dtypes), dtype="object")
+        result_dict = dict(self._dtypes)
+        if cudf.get_option("mode.pandas_compatible"):
+            for key, value in result_dict.items():
+                if isinstance(
+                    value, (cudf.CategoricalDtype, cudf.IntervalDtype)
+                ):
+                    # Convert to pandas dtype for compatibility
+                    result_dict[key] = value.to_pandas()
+                elif isinstance(value, cudf.ListDtype):
```

Comment:
Good to include `StructDtype` and the decimal types as well.

Could you also import them at the top in `from cudf.core.dtypes`?

Reply:
Yup, done.